### PR TITLE
SLING-10011 - Use javax.jcr.Item.getParent to retrieve parent resource in JcrResourceProvider

### PR DESCRIPTION
**Pull request by smiroslav**
https://issues.apache.org/jira/browse/SLING-10011

### CODE REVIEW

File: src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrResourceProvider.java [hunk 1]
```diff
@@ -358,8 +358,8 @@ public Resource getResource(ResolveContext<JcrProviderState> ctx, String path, R
                 if (version == null) {
                     String parentPath = ResourceUtil.getParent(child.getPath());
                     if (parentPath != null) {
-                        Item parentItem = ctx.getProviderState().getResourceFactory()
-                            .getItemOrNull(parentPath);
+                        Item parentItem = ((JcrItemResource)child).getItem().getParent();
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrResourceProvider.java:361]: getItem() as well as getParent might throw an exception - and will not return null, so I think the existing code is correct

**Comment by joerghoh** [comment]
@smiroslav can you please rebase your branch to the current master?

**Comment by smiroslav** [comment]
@joerghoh thanks for looking into this. I have rebased the changes and pushed updated branch.

**Comment by joerghoh** [comment]
As we have @cziegeler's ok, let's move on.

**Comment by cziegeler** [comment]
@joerghoh  Not sure I ever gave an "ok"

**Comment by joerghoh** [comment]
@cziegeler then I misunderstood https://github.com/apache/sling-org-apache-sling-jcr-resource/pull/11#discussion_r546400663

**Comment by cziegeler** [comment]
ok, the code before this PR (== existing code) is correct/works - while with this PR the logic is changed as the methods then used getItem() or getParent() can throw a RepositoryException and they never return "null" - whereas the existing method used does not throw and can return null

**Comment by joerghoh** [comment]
@smiroslav can you raise a new PR for this? Then we need to continue there.

Sorry for the noise :-(

**Comment by smiroslav** [comment]
@cziegeler with the proposed change if RepositoryException is thrown it will be caught by the [existing](https://github.com/apache/sling-org-apache-sling-jcr-resource/blob/org.apache.sling.jcr.resource-3.0.22/src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrResourceProvider.java#L371-L374) catch block, and with that behaviour of the method JcrResourceProvider#getParent stays the same. Did I miss something?

**Comment by cziegeler** [comment]
The difference is at least the warning logged by the catch block  which today doesn't happen

**Comment by smiroslav** [comment]
True. We agree that [contract](https://sling.apache.org/apidocs/sling9/org/apache/sling/spi/resource/provider/ResourceProvider.html#getParent-org.apache.sling.spi.resource.provider.ResolveContext-org.apache.sling.api.resource.Resource-) is not violated? 
In the new PR I can delete waring log message.

**Comment by cziegeler** [comment]
I agree it's the same functionality and behaviour if the warn is not logged for these calls.
Whether this change is an improvement or not is a different topic but I guess there has been enough discussion around this already

**Comment by smiroslav** [comment]
@joerghoh @cziegeler I have created new PR #17.

**Pull request merged**